ichnaea/conftest.py

```diff
@@ -522,3 +525,21 @@ def backoff_sleep_mock():
     """
     with mock.patch("backoff._sync.time.sleep") as mocked_sleep:
         yield mocked_sleep
+
+
+class SingleLogCapture(LogCapture):
+    """Capturing processor with helper method for single log entries."""
+
+    @property
+    def entry(self):
+        """Assert there is a single log entry and return it."""
+        assert len(self.entries) == 1
+        return self.entries[0]
```

Comment:
This is interesting. Is it never the case that the code you're testing emits multiple log lines?

Reply:
I see other cases so ignore my question.

Mmm... This feels like an assertion on how many entries were emitted in addition to accessing the entry. I think maybe I'd rename this to something different. Maybe `only_entry`? Or `there_can_be_only_one_entry`?

I'm spit-balling. It reads weird to me, but I could get used to it.

Reply:
So far, there's exactly one structured log per web request. There are zero logs for static assets and static-like views. I think we lose a lot of the benefits of structured logging if we emit multiple logs per request. So, I like the assertion that there is just one log (most of the time).

This may change when we add structured logging to tasks, which I expect to emit a log per completed task. Or, maybe it won't - this code seems to have avoided ``CELERY_ALWAYS_EAGER`` and the weirdness that causes for Django code. But, if we get into a situation where we have one web log and multiple task logs, does it become ``web_entry``, or ``only_web_entry``? I hate not knowing the future.

I was going to stick with ``entry`` until then, because ``log.entry`` reads better to me than ``log.only_entry``. But because you mentioned it, I've been thinking about it a lot. Then I grepped the code for entry, and got a lot of Sentry code. That will make it even harder to rename in a hypothetical future. So, that's two votes for a longer string.